spk/ffsync/Makefile

```diff
@@ -1,47 +1,55 @@
 SPK_NAME = ffsync
-SPK_VERS = 1.5.2
-SPK_REV = 3
+SPK_VERS = 0.14.1
+SPK_REV = 4
 SPK_ICON = src/ffsync.png
-BETA = 1
 
-BUILD_DEPENDS = cross/python2 cross/setuptools_py2 cross/pip_py2 cross/wheel
-BUILD_DEPENDS += cross/gevent cross/greenlet cross/cffi
+BUILD_DEPENDS = cross/curl cross/mysql-connector-c
+DEPENDS     = cross/syncstorage-rs
 
-DEPENDS     = cross/busybox
-WHEELS      = src/requirements.txt
+PYTHON_PACKAGE = python311
+WHEELS      = src/requirements-crossenv.txt src/requirements-pure.txt
+
+REQUIRED_MIN_DSM = 6.0
+SPK_DEPENDS = MariaDB10:$(PYTHON_PACKAGE)
 
 MAINTAINER   = SynoCommunity
 DESCRIPTION  = Firefox Sync Server 1.5, used for Firefox 29 and later. You can use Firefox Sync Server to synchronize your bookmarks, passwords, settings, history, add-ons and tabs with Firefox on other computers. The service runs on port 8132.
 DISPLAY_NAME = Firefox Sync Server 1.5
-CHANGELOG = "1. Fix Python requirement<br>2. Fix startup issue when pidfile is not yet written"
+CHANGELOG = "1. Upgrade to Mozilla Sync Storage built with Rust."
 
-HOMEPAGE     = https://docs.services.mozilla.com/howtos/run-sync-1.5.html
-LICENSE      = MPL/GPL/LGPL
+HOMEPAGE     = https://mozilla-services.readthedocs.io/en/latest/howtos/run-sync-1.5.html
+LICENSE      = MPL-2.0
 HELPURL      = https://github.com/SynoCommunity/spksrc/wiki/Firefox-Sync-Server-1.5
 
-INSTALLER_SCRIPT = src/installer.sh
-SSS_SCRIPT       = src/dsm-control.sh
-FWPORTS          = src/${SPK_NAME}.sc
+WIZARDS_DIR = src/wizard/
+
+STARTABLE = yes
+SERVICE_USER = auto
+SERVICE_SETUP = src/service-setup.sh
+SERVICE_PORT = 8132
+SERVICE_PORT_TITLE = $(DISPLAY_NAME)
+
+# Admin link for in DSM UI
+ADMIN_PORT = $(SERVICE_PORT)
+
 CONF_DIR         = src/conf/
 
-INSTALL_DEP_SERVICES = mysql
-START_DEP_SERVICES = mysql
+# [mysqlclient]
+DEPENDS += cross/mysql-connector-c cross/mariadb-connector-c
+ENV += MYSQLCLIENT_CFLAGS="$(CFLAGS) -I$(STAGING_INSTALL_PREFIX)/include/mysql -I$(STAGING_INSTALL_PREFIX)/include/mariadb -I$(STAGING_INSTALL_PREFIX)/$(PYTHON_INC_DIR)"
+ENV += MYSQLCLIENT_LDFLAGS="$(LDFLAGS)"
 
-INSTALL_PREFIX = /usr/local/$(SPK_NAME)
+# [bcrypt] and [cryptography]
+# Mandatory for rustc wheel building
+ENV += PYO3_CROSS_LIB_DIR=$(STAGING_INSTALL_PREFIX)/lib/
+ENV += PYO3_CROSS_INCLUDE_DIR=$(STAGING_INSTALL_PREFIX)/include/
+ENV += CRYPTOGRAPHY_DONT_BUILD_RUST=1
```

Comment:
Just curious, have you tried without and confirm if it does build the rust version or not?

Reply:
I'm not sure. I assume you are referring to the cryptography line? Or do you mean the pyo3 lines?

Reply:
Cryptography

Reply:
I'm not sure if I can answer that exactly. As a test, I have a run (https://github.com/SynoCommunity/spksrc/actions/runs/7040991730) which includes this code block and another (https://github.com/SynoCommunity/spksrc/actions/runs/7041348643) which has it completely removed. Checking for lines in the raw logs that contain the word `Cryptography` reveals 17 of the exact same lines with the exception of an extra two in the first run with that ENV variable. These extra lines occur after all the previous matching lines so I don't think it made a difference in what was built. Furthermore, the failure to build `pyo3 v0.14.5` seems to happen the same way.